### PR TITLE
Wherein I Grump at Bad Code

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Just...why.

### CODE REVIEW

**Comment by mattroks101** [comment]
2013 was a rough year.

File: code/game/objects/effects/anomalies.dm [hunk 1]
```diff
@@ -31,10 +31,13 @@
...
-	new /obj/effect/effect/bad_smoke(loc)
+	var/turf/T = get_turf(src)
 
-	for(var/atom/movable/O in src)
-		O.loc = src.loc
+	new /obj/effect/effect/bad_smoke(T)
+
+	if(aSignal)
+		aSignal.forceMove(T)
+		aSignal = null
```
Comment (Crazylemon64) [inline code/game/objects/effects/anomalies.dm]: With `aSignal` nulled out here, sticking `QDEL_NULL` back into `Destroy` might be wise, so an anomaly killed in a manner where it's not neutralized won't leave the beacon as a mess for something else to clean up

Comment (Fox-McCloud) [inline code/game/objects/effects/anomalies.dm]: Or I could make it so when an anomaly is deleted by any means, it just moves the core out.

🤔 

not sure which method to go with.

Comment (Crazylemon64) [inline code/game/objects/effects/anomalies.dm]: I don't think that'd be a good idea, object deletion shouldn't leave a trace, and what of the case if the anomaly "Goes off" and fails to be defused?

**Pull request merged**